OpenRA.Game/Traits/Player/PlayerResources.cs

```diff
@@ -105,6 +105,7 @@ public void GiveCash(int num)
 					checked
 					{
 						Cash += num;
+						if (Cash < 0) Cash = 0;
```

Comment:
please DONT change the cash value! (leave it AS IS!
use this;
```
if (Cash > 0)
    Cash += num;
```
same in bottom case for Earned!

Reply:
But wouldn't the cash go into negative if `num == -(Cash+1)`?

Reply:
Also, one more thing. The OverflowException, is already taken care of. The game can't crash because the entire code is in try/check block.

The worst case scenario you get cash and Earned set to int.max.

Reply:
I think TakeCash already checks for such situtations. So i think this should be fixed by making /givecash all use TakeCash instead of GiveCash when a negative value is given as an argument.

Reply:
So how about wrapping 'TakeCash' and 'GiveCash' in method that decides which to use, based on sign of the argument?

Reply:
Sure, check the usages of both methods and if the dev command is the only one that can do both just do a check which to use in there.